python/pyarrow/tests/conftest.py

```diff
@@ -311,3 +320,23 @@ def s3_server(s3_connection):
         finally:
             if proc is not None:
                 proc.kill()
+
+
+@pytest.fixture(scope='session')
+def gcs_server():
+    port = find_free_port()
+    env = os.environ.copy()
+    args = [sys.executable, '-m', 'testbench', '--port', str(port)]
+    proc = None
+    try:
+        proc = subprocess.Popen(args, env=env)
+    except OSError:
+        pytest.skip('`gcs testbench` raised an error when executing')
```

Comment:
Can we display the actual command line and error here?
```suggestion
    except OSError as e:
        pytest.skip(f"Command {args} failed to execute: {e}")
```